crypto/keys/keys_test.go

```diff
@@ -138,21 +139,19 @@ func TestLoadPrivateKeyAndSign(t *testing.T) {
 		switch {
 		case test.keyPEM != "":
 			k, err = NewFromPrivatePEM(test.keyPEM, test.keyPass)
-			switch gotErr := err != nil; {
-			case gotErr != test.wantLoadErr:
-				t.Errorf("%s: NewFromPrivatePEM(_, _) = (%v, %v), want err? %t", test.name, k, err, test.wantLoadErr)
+			if got, want := errors.ErrorCode(err), test.wantLoadErr; got != want {
+				t.Errorf("%s: NewFromPrivatePEM(_, _) = (_, %v: %v), want err.Code == %v", test.name, got, err, want)
 				continue
-			case gotErr:
+			} else if got != errors.OK {
 				continue
 			}
 
 		case test.keyPath != "":
 			k, err = NewFromPrivatePEMFile(test.keyPath, test.keyPass)
-			switch gotErr := err != nil; {
-			case gotErr != test.wantLoadErr:
-				t.Errorf("%s: NewFromPrivatePEMFile(_, _) = (%v, %v), want err? %t", test.name, k, err, test.wantLoadErr)
+			if got, want := errors.ErrorCode(err), test.wantLoadErr; got != want {
+				t.Errorf("%s: NewFromPrivatePEMFile(_, _) = (_, %v: %v), want err.Code == %v", test.name, got, err, want)
 				continue
-			case gotErr:
+			} else if got != errors.OK {
 				continue
 			}
 
```

Comment:
Sorry, not sure why I added this so far away. I meant line 158 (which Github definitely won't let me add comments on):

```go
default:
  t.Errorf("%s: No PEM or file path set in test definition", test.name)
  continue
```